packages/gatsby-source-wordpress/src/steps/create-schema-customization/helpers.js

```diff
@@ -20,40 +64,66 @@ export const buildTypeName = name => {
     return name
   }
 
+  // Gatsby makes the same type, so we need to rename it to prevent conflicts
   if (name === `Filter`) {
     name = `FilterType`
   }
 
+  if (
+    // Starting in WPGraphQL 1.13.0, Gatsby and WPGraphQL both generate types ending in these strings for every node type in the schema, so we need to rename types to prevent conflicts.
+    // for users on older versions of WPGraphQL we should try to keep the schema how it was before
+    isWpgqlOneThirteenZeroPlus() &&
+    (name.endsWith(`Connection`) ||
+      name.endsWith(`Edge`) ||
+      name.endsWith(`PageInfo`))
+  ) {
+    name += `Type`
+  }
+
   if (name.startsWith(prefix)) {
     return name
   }
 
   return prefix + name
 }
 
-/**
- * Find the first type name of a Type definition pulled via introspection
- * @param {object} type
- */
-export const findTypeName = type =>
-  type?.name ||
-  type?.ofType?.name ||
-  type?.ofType?.ofType?.name ||
-  type?.ofType?.ofType?.ofType?.name
-
 /**
  * Find the first type kind of a Type definition pulled via introspection
  * @param {object} type
  */
-export const findTypeKind = type =>
-  type?.kind ||
-  type?.ofType?.kind ||
-  type?.ofType?.ofType?.kind ||
-  type?.ofType?.ofType?.ofType?.kind
+export const findTypeKind = type => {
+  if (type?.kind) {
+    return type.kind
+  }
+
+  if (type?.ofType) {
+    return findTypeKind(type.ofType)
+  }
+
+  return null
+}
+
+export const findNamedType = type => {
+  if (!type) {
+    return null
+  }
+
+  if (type.ofType) {
+    return findNamedType(type.ofType)
+  }
+
+  return type
+}
+
+export const findNamedTypeName = type => {
+  const namedType = findNamedType(type)
+
+  return namedType?.name
+}
```

Comment:
This refactor is better because the helpers here can get the proper values from types with any depth. This was needed for some new WPGraphQL types.